lib/buffy/throttle_and_timed.ex

```diff
@@ -345,23 +386,57 @@ defmodule Buffy.ThrottleAndTimed do
       @spec handle_continue(do_work :: atom(), Buffy.ThrottleAndTimed.state()) ::
               {:noreply, Buffy.ThrottleAndTimed.state()} | {:noreply, Buffy.ThrottleAndTimed.state(), timeout()}
       def handle_continue(:do_work, %{key: key, args: args} = state) do
-        :telemetry.span(
-          [:buffy, :throttle, :handle],
-          %{args: args, key: key, module: __MODULE__},
-          fn ->
-            result = handle_throttle(args)
-            {result, %{args: args, key: key, module: __MODULE__, result: result}}
-          end
-        )
-
-        new_state = %{state | timer_ref: nil}
-        {:noreply, new_state, unquote(loop_interval)}
+        result =
+          :telemetry.span(
+            [:buffy, :throttle, :handle],
+            %{args: args, key: key, module: __MODULE__},
+            fn ->
+              result = handle_throttle(args)
+              {result, %{args: args, key: key, module: __MODULE__, result: result}}
+            end
+          )
+
+        new_state = %{update_state_with_work_result(state, result) | timer_ref: nil}
+
+        if unquote(loop_interval) do
+          {:noreply, new_state, unquote(loop_interval)}
+        else
+          {:noreply, new_state}
+        end
```

Comment:
If we default to infinity we could simplify this a bit:

```suggestion
        {:noreply, new_state, unquote(loop_interval)}
```

Reply:
Done in 4d99afc